src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+
+            if (nextByte == '0')
+            {
+                ConsumeNumberResult result = ConsumeZero(ref data, ref i);
+                if (result == ConsumeNumberResult.NeedMoreData)
+                    return false;
+                if (result == ConsumeNumberResult.Success)
+                    goto Done;
+
+                nextByte = data[i];
+            }
+            else
+            {
+                i++;
+                ConsumeNumberResult result = ConsumeIntegerDigits(ref data, ref i);
+                if (result == ConsumeNumberResult.NeedMoreData)
+                    return false;
+                if (result == ConsumeNumberResult.Success)
+                    goto Done;
+
+                nextByte = data[i];
+                if (nextByte != '.' && nextByte != 'E' && nextByte != 'e')
+                {
+                    _lineBytePosition += i;
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedNextDigitComponentNotFound, nextByte);
+                }
+            }
+
+            Debug.Assert(nextByte == '.' || nextByte == 'E' || nextByte == 'e');
+
+            if (nextByte == '.')
+            {
+                i++;
+                ConsumeNumberResult result = ConsumeDecimalDigits(ref data, ref i, nextByte);
+                if (result == ConsumeNumberResult.NeedMoreData)
+                    return false;
+                if (result == ConsumeNumberResult.Success)
+                    goto Done;
+
+                nextByte = data[i];
+                if (nextByte != 'E' && nextByte != 'e')
+                {
+                    _lineBytePosition += i;
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedNextDigitEValueNotFound, nextByte);
+                }
+            }
+
+            Debug.Assert(nextByte == 'E' || nextByte == 'e');
+            i++;
+
+            signResult = ConsumeSign(ref data, ref i, nextByte);
+            if (signResult == ConsumeNumberResult.NeedMoreData)
+                return false;
+
+            Debug.Assert(signResult == ConsumeNumberResult.OperationIncomplete);
+
+            i++;
+            ConsumeNumberResult resultExponent = ConsumeIntegerDigits(ref data, ref i);
+            if (resultExponent == ConsumeNumberResult.NeedMoreData)
+                return false;
+            if (resultExponent == ConsumeNumberResult.Success)
+                goto Done;
+            if (resultExponent == ConsumeNumberResult.OperationIncomplete)
+            {
+                _lineBytePosition += i;
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedEndOfDigitNotFound, nextByte);
+            }
+
+        Done:
+            ValueSpan = data.Slice(0, i);
+            consumed = i;
+            return true;
+        }
+
+        private ConsumeNumberResult ConsumeNegativeSign(ref ReadOnlySpan<byte> data, ref int i)
```

Comment:
This method only has one caller, and it uses the result as a boolean.  Should this just return bool?

Reply:
Sure, I thought about that, but since I was using the `enum` elsewhere already, I wanted to keep the structure of all the `ConsumeAComponentOfANumber` similar. I can change it.

Reply:
On second though, I prefer the consistency here.